app/models/book.rb

```diff
@@ -0,0 +1,6 @@
+class Book < ApplicationRecord
+  validates_presence_of :title, :language, :genre
+  has_many :bookshelf_items, dependent: :destroy
+  has_many :reviews, dependent: :destroy
+  has_many :bookshelves, through: :bookshelf_items
+end
```

Comment:
I'm just asking about the below association, do we need to include it?
has_many :users, through: :reviews